File: org.summer.view.widget/src/org/summer/view/widget/internal/ObjectRef.java
package org.summer.view.widget.internal;

import org.summer.view.widget.DependencyObject;

//#region ObjectRef 
    /// <summary> Abstract Object reference. </summary>
    /*internal*/public abstract class ObjectRef 
    { 
        //-----------------------------------------------------
        // 
        //  Constructors
        //
        //-----------------------------------------------------
 
        /// <summary> Constructor is protected - you can only create subclasses. </summary>
        protected ObjectRef() {} 
 
        //------------------------------------------------------
        // 
        //  Public Methods
        //
        //-----------------------------------------------------
 
        /// <summary> Returns the referenced Object. </summary>
        /// <param name="d">Element defining context for the reference. </param> 
        /// <param name="args">See ObjectRefArgs </param> 
        /*internal*/ /*virtual*/ Object GetObject(DependencyObject d, ObjectRefArgs args)
        { 
            return null;
        }

        /// <summary> Returns the data Object associated with the referenced Object. 
        /// Often this is the same as the referenced Object.
        /// </summary> 
        /// <param name="d">Element defining context for the reference. </param> 
        /// <param name="args">See ObjectRefArgs </param>
        /*internal*/ /*virtual*/ public Object GetDataObject(DependencyObject d, ObjectRefArgs args) 
        {
            return GetObject(d, args);
        }
 
        /// <summary> true if the ObjectRef really needs the tree context </summary>
        /*internal*/ public boolean TreeContextIsRequired(DependencyObject target) 
        { 
            return ProtectedTreeContextIsRequired(target);
        } 

        /// <summary> true if the ObjectRef really needs the tree context </summary>
        protected /*virtual*/ boolean ProtectedTreeContextIsRequired(DependencyObject target)
        { 
            return false;
        } 
 
        /// <summary>
        /// true if the ObjectRef uses the mentor of the target element, 
        /// rather than the target element itself.
        /// </summary>
        /*internal*/ public boolean UsesMentor
        { 
            get { return ProtectedUsesMentor; }
        } 
 
        /// <summary>
        /// true if the ObjectRef uses the mentor of the target element, 
        /// rather than the target element itself.
        /// </summary>
        protected /*virtual*/ boolean ProtectedUsesMentor
        { 
            get { return true; }
        } 
 
        /// <summary>
        /// identify this ObjectRef to the user - used by extended tracing 
        /// </summary>
        /*internal*/ abstract String Identify();
    }
 
//#endregion ObjectRef

File: summer/WebContent/js/MS/internal/data/ObjectRef.js
/**
 * Second Check 12-27
 * ObjectRef
 */
/// <summary> Abstract object reference. </summary>
define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	var ObjectRef = declare("ObjectRef", null,{
		constructor:function( ){

		},
		 
        /// <summary> Returns the referenced object. </summary>
        /// <param name="d">Element defining context for the reference. </param> 
        /// <param name="args">See ObjectRefArgs </param> 
        /*internal virtual object */
		GetObject:function(/*DependencyObject*/ d, /*ObjectRefArgs*/ args)
        { 
            return null;
        },

        /// <summary> Returns the data object associated with the referenced object. 
        /// Often this is the same as the referenced object.
        /// </summary> 
        /// <param name="d">Element defining context for the reference. </param> 
        /// <param name="args">See ObjectRefArgs </param>
        /*internal virtual object */
        GetDataObject:function(/*DependencyObject*/ d, /*ObjectRefArgs*/ args) 
        {
            return this.GetObject(d, args);
        },
 
        /// <summary> true if the ObjectRef really needs the tree context </summary>
        /*internal bool */
        TreeContextIsRequired:function(/*DependencyObject*/ target) 
        { 
            return this.ProtectedTreeContextIsRequired(target);
        }, 

        /// <summary> true if the ObjectRef really needs the tree context </summary>
        /*protected virtual bool */
        ProtectedTreeContextIsRequired:function(/*DependencyObject*/ target)
        { 
            return false;
        },
 
 
        /// <summary>
        /// identify this ObjectRef to the user - used by extended tracing 
        /// </summary>
        /*internal abstract string */
        Identify:function(){
        	
        }
	});
	
	Object.defineProperties(ObjectRef.prototype,{
	       /// <summary>
        /// true if the ObjectRef uses the mentor of the target element, 
        /// rather than the target element itself.
        /// </summary>
        /*internal bool */
		UsesMentor:
        { 
            get:function() { return this.ProtectedUsesMentor; }
        }, 
 
        /// <summary>
        /// true if the ObjectRef uses the mentor of the target element, 
        /// rather than the target element itself.
        /// </summary>
        /*protected virtual bool*/ 
        ProtectedUsesMentor:
        { 
            get:function() { return true; }
        } 
 
	});
	
	ObjectRef.Type = new Type("ObjectRef", ObjectRef, [Object.Type]);
	return ObjectRef;
});
